methods.py

```diff
@@ -1549,6 +1549,33 @@ def detect_visual_c_compiler_version(tools_env):
             vc_chosen_compiler_index = vc_x86_amd64_compiler_detection_index
             vc_chosen_compiler_str = "x86_amd64"
 
+    # Newer versions have a different path available
+    vc_amd64_compiler_detection_index = tools_env["PATH"].upper().find(tools_env['VCTOOLSINSTALLDIR'].upper() + "BIN\\HOSTX64\\X64;")
```

Comment:
This fails if `VCTOOLSINSTALLDIR` is not set, which is the case for versions before 2017. Add the changes from 8b01f9d28ffccb21db04867d1265f0be9d40aa53 to make it work properly.